### PR TITLE
[ticket=no][risk=no] Fix bug where workspace share from workspace card was missing an access level

### PR DESCRIPTION
**Pull request by ericsong**
I did some clean up while I was poking around. We were loading profile from various components unnecessarily.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-share.tsx [hunk 1]
```diff
@@ -323,6 +322,7 @@ export const WorkspaceShare = fp.flow(withCurrentWorkspace(), withUserProfile())
   }
 
   get hasPermission(): boolean {
+    console.log(this.props.workspace);
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-share.tsx:325]: rm

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 2]
```diff
@@ -179,30 +176,12 @@ export class WorkspaceCard extends React.Component<WorkspaceCardProps, Workspace
     }
   }
 
-  // The function called when the 'share' action is called on a workspace card
-  // within the recentWorkspaces list.
-  async handleShareAction() {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:184]: so this was all duplicated in WorkspaceShare already?

**Review by jmthibault79 — APPROVED** [review]
Nice cleanup!

Tested locally: sharing works from the sidebar as well as the recent-workspaces and all-workspaces pages.  I didn't see a good way to test workspace-library.

File: ui/src/app/pages/workspace/workspace-card.tsx (hunk 2 above)
Comment (ericsong) [reply ui/src/app/pages/workspace/workspace-card.tsx:184]: yep, but only recently. I moved this login into the WorkspaceShare component and didn't realize there were other places where that action was available and had stuff to be cleaned up.

**Comment by ericsong** [comment]
Merging before CircleCI since the only new commit was a removal of `console.log()`. Tests were green before that commit.

**Pull request merged**